src/layers.c

```diff
@@ -131,22 +145,36 @@ surface_commit_notify(struct wl_listener *listener, void *data)
 		return;
 	}
 
+	struct output *output = (struct output *)layer_surface->output->data;
 	if (layer_surface->current.committed
 			|| layer->mapped != layer_surface->mapped) {
 		layer->mapped = layer_surface->mapped;
-		struct output *output =
-			output_from_wlr_output(layer->server, wlr_output);
 		layers_arrange(output);
 	}
+
+	/*
+	 * We need to call cursor_update_focus() to get a pointer-enter event
+	 * when a new layer surface appears under the cursor.  This enables
+	 * gdk_device_get_window_at_position() to work without moving pointer.
+	 * See issue #667
+	 */
+	cursor_update_focus(output->server);
```

Comment:
This will get called every time the client updates the buffer. Maybe it could be moved within the above condition so its only called on the first map?